### ISSUE TITLE
The specified "childViewContainer" was not found bug

### ISSUE THREAD

**Issue by jagin** [marionettejs/backbone.marionette#2436]
Hi,

I'm using Marionette 

I have a view like this:

```javascript
    ContentListView = Marionette.CompositeView.extend({
        childView: ContentItemView,
        childViewContainer: "tbody",
        emptyView: ContentEmptyView,
        template: contentListJst,
        className: "content-list-view",
        triggers: {
            "click .js-add": "add:clicked",
        }
    });
```

The `contentListJst` template contains `<tbody</tbody>` tag as it should.

I create the view with empty collection like this:
```javascript
    var contentsView = new ContentsView({
        collection: new Backbone.Collection()
    });
```
next, without rendering the view i'm destroying it with:
```javascript
    contentsView.destroy();
```

and got the error:
```
ChildViewContainerMissingError: The specified "childViewContainer" was not found: tbody
var error = Error.call(this, message);
backbon...ette.js (wiersz 893)
```

The stack trace is  as follow:
```
Marionette.CompositeView.Marionette.CollectionView.extend.getChildViewContainer (backbone.marionette.js:2426)
Marionette.CompositeView.Marionette.CollectionView.extend._insertAfter (backbone.marionette.js:2394)
Marionette.CollectionView.Marionette.View.extend.attachHtml (backbone.marionette.js:2161)
Marionette.CollectionView.Marionette.View.extend.renderChildView (backbone.marionette.js:2083)
Marionette.CollectionView.Marionette.View.extend.addEmptyView (backbone.marionette.js:1986)
Marionette.CollectionView.Marionette.View.extend.showEmptyView (backbone.marionette.js:1929)
Marionette.CollectionView.Marionette.View.extend.checkEmpty (backbone.marionette.js:2129)
Marionette.CollectionView.Marionette.View.extend.destroyChildren (backbone.marionette.js:2213)
Marionette.CollectionView.Marionette.View.extend.destroy (backbone.marionette.js:2202)
```

It looks like even though the view was not rendered at all the `destroy` function is trying to attach some HTML.

**Comment by paulfalgout** [comment]
Confirmed this bug.. drop this in a console:

```js
var ContentListView = Marionette.CompositeView.extend({
        childView: Marionette.ItemView.extend({ template: _.template('<div></div>') }),
        childViewContainer: "tbody",
        emptyView: Marionette.ItemView.extend({ template: _.template('<div></div>') }),
        template: _.template('<table><tbody></tbody></table>'),
        className: "content-list-view",
        triggers: {
            "click .js-add": "add:clicked",
        }
});
var contentsView = new ContentListView({
    collection: new Backbone.Collection()
});
contentsView.destroy();
```

The stack pretty well explains what is happening and it appears to be a very old regression.

It may be fixed by just preventing destroyChildren if the view was never rendered.

**Comment by jasonLaster** [comment]
Thanks

**Comment by ahumphreys87** [comment]
hey @jagin this is caused by `destroyChildren` calling `checkEmpty` unnecessarily and has been fixed via #2384. This will be released with 2.4.2
